src/components/controls/selects/MultiSelect/MultiSelectInput.tsx

```diff
@@ -0,0 +1,469 @@
+  font-size: ${({ theme }) => theme.fontSizes.md};
+  font-weight: ${({ theme }) => theme.fontWeights.book};
+  line-height: calc(4 / 3);
+  background: ${({ theme }) => theme.colors.background.secondary};
+  color: ${({ theme }) => theme.colors.content.primary};
+  inline-size: 100%;
+  padding-block: ${({ theme }) => theme.spacingBase}rem;
+  padding-inline-end: ${({ theme }) => theme.spacingBase}rem;
+  border-radius: ${({ hasLabel, theme }) =>
+    hasLabel ? theme.borderRadii.sm : theme.borderRadii.xl};
+
+  &:focus-within {
+    box-shadow: 0px 0px 0px 1px ${({ theme }) => theme.colors.border.selected} inset;
+    outline: none;
+  }
+
+  &:hover:not(:disabled) {
+    background: ${({ theme }) => theme.colors.background.tertiary};
+  }
+
+  &::placeholder {
+    color: ${({ theme }) => theme.colors.content.secondary};
+  }
+
+  &:disabled,
+  &:disabled::placeholder {
+    color: ${({ theme }) => theme.colors.content.tertiary};
+    cursor: not-allowed;
+  }
+`
+
+const Scroller = styled.div`
+  overflow: auto;
+  scrollbar-width: none;
+  flex: 1 1 auto;
+  padding-inline-start: ${({ theme }) => 2 * theme.spacingBase}rem;
+  display: flex;
+
+  &::-webkit-scrollbar {
+    display: none;
+  }
+`
+
+const ScrollContent = styled.div`
+  display: flex;
+  min-inline-size: 100%;
+  align-items: center;
+  gap: ${({ theme }) => theme.spacingBase}rem;
+`
+
+const InputWrapper = styled.div`
+  position: relative;
+  flex: 1 1 auto;
+  inline-size: 100%;
+  transition: min-inline-size 0.1s;
+  block-size: ${({ theme }) => 4 * theme.spacingBase}rem;
+  line-height: ${({ theme }) => 4 * theme.spacingBase}rem;
+`
+
+const Shadow = styled.span`
+  // add some error margin
+  padding-inline-end: ${({ theme }) => theme.spacingBase}rem;
+  white-space: pre;
+  opacity: 0;
+  visibility: hidden;
+  position: relative;
+  z-index: -1;
+`
+
+const Input = styled.input<SelectionProps>`
+  background: transparent;
+  position: absolute;
+  inset: 0;
+  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
+  ${({ isSelected }) => (!isSelected ? "caret-color: transparent;" : "")};
+
+  &:focus {
+    outline: 0;
+  }
+`
+
+const Pill = styled.button<SelectionProps>`
+  background-color: ${({ theme, isSelected }) =>
+    isSelected ? theme.colors.background.primaryInverted : theme.colors.background.tertiary};
+  ${({ theme, isSelected }) =>
+    isSelected ? `color: ${theme.colors.content.primaryInverted}` : ""};
+  border-radius: ${({ theme }) => theme.borderRadii.sm};
+  padding: ${({ theme }) => 0.5 * theme.spacingBase}rem ${({ theme }) => theme.spacingBase}rem;
+  white-space: nowrap;
+  overflow: hidden;
+
+  p {
+    overflow: hidden;
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    margin: 0;
+  }
+`
+
+const SelectedOptionsWrapper = styled.div<WrapperProps>`
+  display: flex;
+  flex: 0 1 auto;
+  gap: ${({ theme }) => theme.spacingBase}rem;
+  margin-inline-start: ${({ theme, hasLabel }) => (hasLabel ? -1 * theme.spacingBase : 0)}rem;
```

Comment:
Pretty sure you can do something like this if you don't want to set `margin-inline-start` when there is no label:

```suggestion
  margin-inline-start: ${({ theme, hasLabel }) => hasLabel && `${theme.spacingBase}rem`};
```